jscomp/core/js_implementation.ml

```diff
@@ -164,9 +164,9 @@ let after_parsing_impl ppf outputprefix (ast : Parsetree.structure) =
         let lambda, exports =
           Translmod.transl_implementation modulename typedtree_coercion
         in
-        let js_program ~module_system =
+        let js_program =
           print_if_pipe ppf Clflags.dump_rawlambda Printlambda.lambda lambda
-          |> Lam_compile_main.compile outputprefix module_system exports
+          |> Lam_compile_main.compile outputprefix NodeJS exports
```

Comment:
Temporarily I removed the argument `module_system` to do eager evaluation here and pass `NodeJS` instead, then tests are passed.
This is not the cure for the root cause, Lam_compile.compile needs to know the module_system info.